### PR TITLE
feat: update title through excel import

### PR DESCRIPTION
**Pull request by yspringfield**
Fixed: #8270

### CODE REVIEW

**Review by GrandSchtroumpf — COMMENTED** [review]
I think we should discuss this feature. It adds more complexity on something already very complexe.

File: libs/import/src/lib/view-extracted-elements/titles/view-extracted-titles.component.html [hunk 1]
```diff
@@ -9,7 +8,8 @@ <h3>Titles to Create</h3>
   </ng-container>
   <ng-container *ngIf="moviesToUpdate$ | async as moviesToUpdate">
     <ng-container *ngIf="moviesToUpdate.data.length">
-      <h3>Titles to Update & Submit</h3>
+      <h3 class="margin-top">Titles to Update & Submit</h3>
```
Comment (GrandSchtroumpf) [inline libs/import/src/lib/view-extracted-elements/titles/view-extracted-titles.component.html:11]: If you've got a h3 you need a article or section container.

File: libs/import/src/lib/view-extracted-elements/titles/view-extracted-titles.component.scss [hunk 2]
```diff
@@ -0,0 +1,3 @@
+.margin-top{
```
Comment (GrandSchtroumpf) [inline libs/import/src/lib/view-extracted-elements/titles/view-extracted-titles.component.scss:1]: You should not target a specific h3 but use a gap or something like that between containers

File: libs/utils/src/lib/utils.ts [hunk 3]
```diff
@@ -302,3 +302,20 @@ export function toLanguageVersionString(languages: LanguageRecord) {
...
+        obj.splice(+k, 1)
+      } else {
+        delete obj[k];
+      }
+    } else if (typeof obj[k] === "object") {
+      removeNulls(obj[k]);
+    }
+  }
+  return obj;
+}
```
Comment (GrandSchtroumpf) [inline libs/utils/src/lib/utils.ts:321]: Better to treat the array in another way than the object
```suggestion
export function removeNulls(obj: any) {
  if (Array.isArray(obj)) return obj.map(removeNulls).filter(el => el !== null && el !== undefined);
  for (const k of Object.keys(obj)) {
    if (obj[k] === null && obj[k] === undefined) {
      removeNulls(obj[k]);
    } else {
      delete obj[k];
    }
  }
  return obj;
}
```

File: libs/utils/src/lib/spreadsheet/import.ts [hunk 4]
```diff
@@ -172,7 +172,7 @@ export interface ValueWithErrorSimple<T = unknown> extends SpreadsheetImportErro
 }
 
 function isValueWithError(o: unknown): o is ValueWithError {
-  return typeof o === 'object' && ('value' in o) && ('error' in o);
+  return o && o instanceof ImportLog;
```
[the remark this answers is not in the document]
Comment (yspringfield) [reply libs/utils/src/lib/spreadsheet/import.ts:175]: Bildschirmfoto 2022-05-03 um 10 27 07 AM
`ValueWithError` is a ts type that does not exists at runtime.

File: libs/import/src/lib/view-extracted-elements/titles/view-extracted-titles.component.scss [hunk 5]
```diff
@@ -0,0 +1,3 @@
+section:nth-child(2) {
+  margin-top: 24px;
+}
```
Comment (GrandSchtroumpf) [inline libs/import/src/lib/view-extracted-elements/titles/view-extracted-titles.component.scss:3]: It's better to have something like that
```suggestion
:host {
  display: flex;
  flex-direction: column;
  gap: 24px;
}
```

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**